crates/anvil/src/eth/miner.rs

```diff
@@ -75,13 +78,13 @@ impl Miner {
         cx: &mut Context<'_>,
     ) -> Poll<Vec<Arc<PoolTransaction>>> {
         self.inner.register(cx);
-        // Add replay transactions to the pool
-        if let Some(replay_transactions) = self.replay_transactions.take() {
-            replay_transactions.into_iter().for_each(|tx| {
-                pool.add_transaction(tx).unwrap();
-            });
+        // Include any forced transactions if they exist
+        if let Some(transactions) = self.force_include_transactions.take() {
+            let next = ready!(self.mode.write().poll(pool, cx));
```

Comment:
this order is problematic

we need to always call ready! first then .take seperately

Reply:
updated